website/models.py

```diff
@@ -658,6 +661,16 @@ def __str__(self):
         return f"Q: {self.question} | A: {self.answer} at {self.timestamp}"
 
 
+class BlockedIP(models.Model):
```

Comment:
It looks like you could use a flag of `IP` model, something like `is_blocked` and toggle it as needed instead of creating duplicates for every blocked IP address.